packages/main/src/Carousel.js

```diff
@@ -0,0 +1,296 @@
+		return litRender;
+	}
+
+	static get styles() {
+		return CarouselCss;
+	}
+
+	static get template() {
+		return CarouselTemplate;
+	}
+
+	static get pageTypeLimit() {
+		return 9;
+	}
+
+	constructor() {
+		super();
+
+		this._scrollEnablement = new ScrollEnablement(this);
+		this._scrollEnablement.attachEvent("touchend", event => {
+			this._updateScrolling(event);
+		});
+
+		this.i18nBundle = getI18nBundle("@ui5/webcomponents");
+	}
+
+	onBeforeRendering() {
+		if (!isDesktop()) {
+			this.itemsPerPage = 1;
+		}
+	}
+
+	onAfterRendering() {
+		this._scrollEnablement.scrollContainer = this.getDomRef();
+	}
+
+	_updateScrolling(event) {
+		if (!event) {
+			return;
+		}
+
+		if (event.isLeft) {
+			this.navigateLeft();
+		} else if (event.isRight) {
+			this.navigateRight();
+		}
+	}
+
+	_onkeydown(event) {
+		if (event.target !== this.getDomRef()) {
+			return;
+		}
+
+		if (isLeft(event) || isDown(event)) {
+			this.navigateLeft();
+		} else if (isRight(event) || isUp(event)) {
+			this.navigateRight();
+		}
+	}
+
+	navigateLeft() {
+		if (this.selectedIndex - 1 < 0) {
+			if (this.cycling) {
+				this.selectedIndex = this.items.length - 1;
+			}
+		} else {
+			--this.selectedIndex;
+		}
+	}
+
+	navigateRight() {
+		if (this.selectedIndex + 1 > this.items.length - 1) {
+			if (this.cycling) {
+				this.selectedIndex = 0;
+			}
+		} else {
+			++this.selectedIndex;
+		}
+	}
+
+	/**
+	 * Assuming that all items have the same width
+	 * @private
+	 */
+	get items() {
+		const result = [],
+			innerArraysLength = Math.ceil(this.content.length / this.itemsPerPage);
+
+		for (let i = 0; i < innerArraysLength; i++) {
+			result.push([]);
+			for (let j = 0; j < this.itemsPerPage; j++) {
+				result[i].push({
+					item: this.content[(i * this.itemsPerPage) + j],
+					tabIndex: i === this.selectedIndex ? "0" : "-1",
+				});
+			}
+		}
+
+		return result;
+	}
+
+	get styles() {
+		return {
+			content: {
+				"left": `-${this.selectedIndex * 100}%`,
+			},
+		};
+	}
+
+	get classes() {
+		return {
+			navigation: {
+				"ui5-carousel-navigation-wrapper": true,
+				"ui5-carousel-navigation-with-buttons": this.arrowsPlacement === CarouselArrowsPlacement.Navigation,
+			},
+			item: {
+				"ui5-carousel-item": true,
+				"ui5-carousel-item-multiple": this.itemsPerPage > 1,
+			},
+		};
+	}
+
+	get isPageTypeDots() {
+		return this.items.length < Carousel.pageTypeLimit;
+	}
+
+	get dots() {
+		return this.items.map((item, index) => {
+			return {
+				active: index === this.selectedIndex,
+			};
+		});
+	}
+
+	get arrows() {
+		return {
+			content: isDesktop() && this.arrowsPlacement === CarouselArrowsPlacement.Content,
+			navigation: isDesktop() && this.arrowsPlacement === CarouselArrowsPlacement.Navigation,
+		};
+	}
+
+	get ofText() {
+		return this.i18nBundle.getText(CAROUSEL_OF_TEXT);
+	}
+
+	get currenlySelectedIndexToShow() {
+		return this.selectedIndex + 1;
+	}
+
+	static async define(...params) {
+		await Promise.all([
```

Comment:
No need for Promise.all, but I don't mind that much (since we can add further dependencies in the future and it will be handier)